ampersand-state.js

```diff
@@ -177,11 +178,12 @@ assign(Base.prototype, Events, {
             }
 
             isEqual = this._getCompareForType(def.type);
+            onChange = this._getOnChangeForType(def.type);
             dataType = this._dataTypes[def.type];
 
             // check type if we have one
             if (dataType && dataType.set) {
-                cast = dataType.set(newVal);
+                cast = dataType.set.call(this, newVal, options);
```

Comment:
Isn't the purpose of the new onChange function where that should happen? Rather than in the set?